### DISCUSSION TITLE
What happens if multiple io_uring instances submit accept SQEs on the same socket listener?

### DISCUSSION THREAD

**Discussion by Kleinmarb** [axboe/liburing#1251] [Q&A] [answered]
Will this result in load balancing like SO_REUSEPORT or post a failed cqe on all instances but one where it will wait for a connection? Maybe something else.

**Comment by axboe** [comment, marked as answer]
If you submit multiple accepts for the same listener, then just one of them will trigger when someone connects. If it's a single shot accept, then that single accept will be done, but the other(s) remain. The next connect will trigger one of the other ones, and so forth.

You'd be better off using multishot accept, where the accept is persistent and will just trigger whenever someone connects. And then if you need to pass that accepted to fd to someone else to handle, that's how you'd do balancing. The persistent multishot accept is more efficient than having a lot of pending accepts.
